internal/tools/cypher/json_tagged_types.go

```diff
@@ -0,0 +1,50 @@
+package cypher
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+type Params map[string]any
+
+func (cp *Params) UnmarshalJSON(data []byte) error {
+	decoder := json.NewDecoder(bytes.NewReader(data))
+
+	decoder.UseNumber()
+
+	var temp map[string]any
+	if err := decoder.Decode(&temp); err != nil {
+		return err
+	}
+
+	*cp = ConvertNumbers(temp).(map[string]any)
```

Comment:
This does not feel right to me at the first glance, why are we doing it?

Reply:
Done the fix for the panic error, 
I'll open the PR, 
The "ConvertNumbers" I left as it was from your PR, if you look "params_converter.go" 
is the same method you have added with the difference for the float handling, but that's not changes the "mutate and return", let's not extend the scope of this improvement. 
https://github.com/neo4j/mcp/pull/104/files/227ce17e3c5d870a5b95f8b1e84b015442c311be#diff-7be318cc07d6bbd792cb31b3d946e940751f2180f75673f4207f73591589e416L54